android/src/main/kotlin/com/mr/flutter/plugin/filepicker/FileUtils.kt

```diff
@@ -178,140 +177,149 @@ object FileUtils {
         }
+        val detector = tika.detector
 
-        private fun getMimeTypeForBytes(fileName: String?, bytes: ByteArray?): String {
-            val tika = Tika()
+        val stream = TikaInputStream.get(bytes)
+        val metadata = Metadata()
+        metadata.set(TikaCoreProperties.RESOURCE_NAME_KEY, fileName)
+        return detector.detect(stream, metadata).toString()
+    }
 
-            if (fileName.isNullOrEmpty()) {
-                return tika.detect(bytes)
-            }
-            val detector = tika.detector
-
-            val stream = TikaInputStream.get(bytes)
-            val metadata = Metadata()
-            metadata.set(TikaCoreProperties.RESOURCE_NAME_KEY, fileName)
-            return detector.detect(stream, metadata).toString()
-        }
-
-        fun FilePickerDelegate.saveFile(
-            fileName: String?,
-            type: String?,
-            initialDirectory: String?,
-            bytes: ByteArray?,
-            result: MethodChannel.Result
-        ) {
-            if (!this.setPendingMethodCallResult(result)) {
-                finishWithAlreadyActiveError(result)
-                return
-            }
-            val intent = Intent(Intent.ACTION_CREATE_DOCUMENT)
-            intent.addCategory(Intent.CATEGORY_OPENABLE)
-            if (!fileName.isNullOrEmpty()) {
-                intent.putExtra(Intent.EXTRA_TITLE, fileName)
-            }
-            this.bytes = bytes
-            if ("dir" != type) {
+    fun FilePickerDelegate.saveFile(
+        fileName: String?,
+        type: String?,
+        initialDirectory: String?,
+        bytes: ByteArray?,
+        result: MethodChannel.Result
+    ) {
+        if (!this.setPendingMethodCallResult(result)) {
+            finishWithAlreadyActiveError(result)
+            return
+        }
+        val intent = Intent(Intent.ACTION_CREATE_DOCUMENT)
+        intent.addCategory(Intent.CATEGORY_OPENABLE)
+        if (!fileName.isNullOrEmpty()) {
+            intent.putExtra(Intent.EXTRA_TITLE, fileName)
+        }
+        this.bytes = bytes
+        if ("dir" != type) {
+            try {
                 intent.type = getMimeTypeForBytes(fileName = fileName, bytes = bytes)
-            }
-            if (!initialDirectory.isNullOrEmpty()) {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                    intent.putExtra(DocumentsContract.EXTRA_INITIAL_URI, initialDirectory.toUri())
-                }
-            }
-            if (intent.resolveActivity(activity.packageManager) != null) {
-                activity.startActivityForResult(intent, SAVE_FILE_CODE)
-            } else {
+            } catch (t: Throwable) {
+                intent.type = "*/*"
                 Log.e(
                     FilePickerDelegate.TAG,
-                    "Can't find a valid activity to handle the request. Make sure you've a file explorer installed."
+                    "Tika failed to detect mime type. Throws: $t"
```

Comment:
I would not mention Tika in the error, because it is an implementation detail.

```suggestion
                    "Failed to detect mime type. $t"
```